### ISSUE TITLE
pylint `unused-variable`

### ISSUE THREAD

**Issue by danstoner** [iDigBio/idb-backend#164]
W0612 unused-variable
Message
'Unused variable %r'

Description
Used when a variable is defined but not used.

For example, there are a number of situations where a variable is declared as part of a for loop but never used.  Most likely those could be replaced with an underscore.

```
    count = 0
    for e in iterable:
        count += 1
    return count
```

should become:

```
    count = 0
    for _ in iterable:
        count += 1
    return count
```

**Comment by danstoner** [comment]
```
idb-backend]$ pylint idb idigbio_ingestion idigbio_workers/ | grep unused-variable
idb/data_tables/build_taxon_index.py:95:8: W0612: Unused variable 'r' (unused-variable)
idb/data_tables/taxon.py:21:4: W0612: Unused variable 'sp_count' (unused-variable)
idb/data_tables/taxon.py:22:4: W0612: Unused variable 'sp_non_count' (unused-variable)
idb/data_tables/taxon.py:23:4: W0612: Unused variable 'fail' (unused-variable)
idb/data_tables/taxon.py:24:4: W0612: Unused variable 'names' (unused-variable)
idb/data_tables/taxon.py:25:4: W0612: Unused variable 'ids' (unused-variable)
idb/data_tables/taxon.py:26:4: W0612: Unused variable 'needs_name' (unused-variable)
idb/data_tables/taxon.py:35:20: W0612: Unused variable 'skiprest' (unused-variable)
idb/data_tables/taxon.py:125:8: W0612: Unused variable 'dr' (unused-variable)
idb/data_tables/taxon_extended.py:21:12: W0612: Unused variable 'data_etag' (unused-variable)
idb/stats/stats_rewriter.py:141:8: W0612: Unused variable 'e' (unused-variable)
idb/helpers/conversions.py:804:12: W0612: Unused variable 'vn' (unused-variable)
idb/helpers/conversions.py:952:4: W0612: Unused variable 'dt_serial' (unused-variable)
idb/helpers/conversions.py:1007:12: W0612: Unused variable 'r' (unused-variable)
idb/helpers/__init__.py:19:8: W0612: Unused variable 'e' (unused-variable)
idb/postgres_backend/db.py:345:8: W0612: Unused variable 'rc1' (unused-variable)
idb/indexing/index_from_postgres.py:311:16: W0612: Unused variable 'ok' (unused-variable)
idb/indexing/index_from_postgres.py:311:20: W0612: Unused variable 'item' (unused-variable)
idb/indexing/index_helper.py:39:11: W0612: Unused variable 'ck' (unused-variable)
idigbio_ingestion/verify_ceph_objects.py:102:12: W0612: Unused variable 'cr' (unused-variable)
idigbio_ingestion/update_publisher_recordset.py:361:4: W0612: Unused variable 'auto_publish' (unused-variable)
idigbio_ingestion/lib/delimited.py:199:12: W0612: Unused variable 'line' (unused-variable)
idigbio_ingestion/lib/fileproxy.py:114:16: W0612: Unused variable 'l' (unused-variable)
idigbio_ingestion/lib/dwca.py:176:8: W0612: Unused variable 'linesplit' (unused-variable)
idigbio_ingestion/lib/dwca.py:209:12: W0612: Unused variable 'next_line' (unused-variable)
idigbio_workers/__init__.py:23:4: W0612: Unused variable 'scheme' (unused-variable)
idigbio_workers/__init__.py:23:24: W0612: Unused variable 'user' (unused-variable)
idigbio_workers/__init__.py:23:30: W0612: Unused variable 'password' (unused-variable)
idigbio_workers/__init__.py:23:46: W0612: Unused variable 'query' (unused-variable)
idigbio_workers/generate_static_datasets.py:87:4: W0612: Unused variable 'u' (unused-variable)
```

**Comment by danstoner** [comment]
Fixed up the low hanging fruit in a3866decca3a4b953482c5bc489d8711a81f9931
